packages/geo/src/lib/filter/ogc-filter-toggle-button/ogc-filter-toggle-button.component.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Component,
+  Input,
+  ChangeDetectorRef,
+  OnInit
+} from '@angular/core';
+
+import {
+  OgcFilterableDataSource,
+  IgoOgcFilterObject,
+  OgcPushButton,
+  OgcPushButtonBundle
+
+} from '../../filter/shared/ogc-filter.interface';
+import { OgcFilterWriter } from '../../filter/shared/ogc-filter';
+import { IgoMap } from '../../map';
+import { OGCFilterService } from '../shared/ogc-filter.service';
+import { WMSDataSource } from '../../datasource/shared/datasources/wms-datasource';
+
+@Component({
+  selector: 'igo-ogc-filter-toggle-button',
+  templateUrl: './ogc-filter-toggle-button.component.html',
+  styleUrls: ['./ogc-filter-toggle-button.component.scss']
+})
+export class OgcFilterToggleButtonComponent implements OnInit {
+  private ogcFilterWriter: OgcFilterWriter;
+  private _dataSource: OgcFilterableDataSource;
+  public color = 'primary';
+  private _map: IgoMap;
+
+  @Input() refreshFilters: () => void;
+
+  @Input()
+  get datasource(): OgcFilterableDataSource {
+    return this._dataSource;
+  }
+  set datasource(value: OgcFilterableDataSource) {
+    this._dataSource = value;
+    this.cdRef.detectChanges();
+  }
+
+  @Input()
+  get map(): IgoMap {
+    return this._map;
+  }
+  set map(value: IgoMap) {
+    this._map = value;
+  }
+
+  public pushButtonBundle: OgcPushButtonBundle[] = [];
+
+  constructor(
+    private ogcFilterService: OGCFilterService,
+    private cdRef: ChangeDetectorRef
+  ) {
+    this.ogcFilterWriter = new OgcFilterWriter();
+  }
+
+  ngOnInit() {
+
+    if (this.datasource.options.ogcFilters &&
+      this.datasource.options.ogcFilters.pushButtons) {
+        this.pushButtonBundle = this.datasource.options.ogcFilters.pushButtons as OgcPushButtonBundle[];
+    }
+    this.applyFilters();
+
+  }
+
+  getToolTip(pb: OgcPushButton) {
+    let tt;
+    if (pb.tooltip) {
+      tt = pb.tooltip;
+    }
+    return tt;
+  }
+
+  setColor(pb: OgcPushButton) {
```

Comment:
This method returns something, is it properly named? Also, return type.

Reply:
rename to getButtonColor